### PR TITLE
Update index.md

### PR DESCRIPTION
**Pull request by Zearin**
Minor markup tweaks and copy-edits.

### Description of the Change

Minor markup tweaks and copy-edits to the main documentation page (`index.md`, which becomes the content for https://mochajs.org).

### Alternate Designs

N/A

### Why should this be in core?

Well...the docs are _already_ part of the main package....

### Benefits

* Minor boost to aesthetics and readability
* Consistent capitalization of “Mocha” (outside of code)

### Possible Drawbacks

I can’t think of any.

### Applicable issues

Does not impact production code.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -658,7 +658,7 @@ You can choose to retry failed tests up to a certain number of times. This featu
 
 This feature does re-run a failed test and its corresponding `beforeEach/afterEach` hooks, but not `before/after` hooks. `this.retries()` has no effect on failing hooks.
 
-**NOTE**: Example below was written using Selenium webdriver (which [overwrites global Mocha hooks][selenium-webdriver-testing] for `Promise` chain).
+**NOTE**: The following example was written using Selenium Webdriver (which [overwrites global Mocha hooks][selenium-webdriver-testing] for `Promise` chain).
```
Comment (geshii) [inline docs/index.md:661]:
```suggestion
**NOTE**: The following example was written using the Selenium Webdriver (which [overwrites global Mocha hooks][selenium-webdriver-testing] for the `Promise` chain).
```

File: docs/index.md [hunk 2]
```diff
@@ -835,7 +835,7 @@ Use this option to have Mocha check for global variables that are leaked while r
 
 > _Updated in v4.0.0._
 
-TL;DR: If your tests hang after an upgrade to Mocha v4.0.0 or newer, use `--exit` for a quick (though not necessarily recommended) fix.
+**TL;DR:** If your tests hang after an upgrade to Mocha v4.0.0 or newer, use `--exit` for a quick (though not necessarily recommended) fix.
```
Comment (geshii) [inline docs/index.md:838]:
```suggestion
**TL;DR:** If your tests are hanging after an upgrade to Mocha v4.0.0 or newer, use `--exit` for a quick (though not necessarily recommended) fix.
```

File: docs/index.md [hunk 3]
```diff
@@ -1166,7 +1166,7 @@ These flags vary depending on your version of Node.js.
 > _New in Node.js v12.12.0_
 
 If the [`--enable-source-maps`](https://nodejs.org/dist/latest-v12.x/docs/api/cli.html#cli_enable_source_maps) flag
-is passed to mocha, source maps will be collected and used to provide accurate stack traces for transpiled code:
+is passed to Mocha, source maps will be collected and used to provide accurate stack traces for transpiled code:
```
Comment (geshii) [inline docs/index.md:1169]: isn't it supposed to be lowercase?

File: docs/index.md [hunk 4]
```diff
@@ -1466,8 +1466,8 @@ Multiple root hook plugins can be registered by using `--require` multiple times
 
 To migrate your tests using root hooks to a root hook plugin:
 
-1. Find your root hooks (hooks defined _outside_ of a suite--usually `describe()` callback).
-1. Create a new file, e.g., `test/hooks.js`.
+1. Find your root hooks (hooks defined _outside_ of a suite&mdash;usually `describe()` callback).
+1. Create a new file (e.g., `test/hooks.js`).
```
Comment (geshii) [inline docs/index.md:1470]:
```suggestion
1. Create a new file (for example, `test/hooks.js`).
```

File: docs/index.md [hunk 5]
```diff
@@ -1571,7 +1571,7 @@ export async function mochaGlobalSetup() {
 
 To use it, load this file when running Mocha via `mocha --require fixtures.cjs` (or whatever you have named the file).
 
-> Remember: you can define "requires" in a [configuration file](#configuring-mocha-nodejs).
+> _Remember:_ You can define "requires" in a [configuration file](#configuring-mocha-nodejs).
```
Comment (geshii) [inline docs/index.md:1574]:
```suggestion
> _Reminder:_ You can define `requires` in a [configuration file](#configuring-mocha-nodejs).
```

File: docs/index.md [hunk 6]
```diff
@@ -2253,15 +2253,15 @@ _Note_: Double quotes around the glob are recommended for portability.
...
-| ERR_MOCHA_UNSUPPORTED            | requested behavior, option, or parameter is unsupported      |
+| Code                               | Description                                                  |
+| ---------------------------------- | ------------------------------------------------------------ |
+| `ERR_MOCHA_INVALID_ARG_TYPE`       | wrong type was passed for a given argument                   |
+| `ERR_MOCHA_INVALID_ARG_VALUE`      | invalid or unsupported value was passed for a given argument |
+| `ERR_MOCHA_INVALID_EXCEPTION`      | a falsy or otherwise underspecified exception was thrown     |
+| `ERR_MOCHA_INVALID_INTERFACE`      | interface specified in options not found                     |
+| `ERR_MOCHA_INVALID_REPORTER`       | reporter specified in options not found                      |
+| `ERR_MOCHA_NO_FILES_MATCH_PATTERN` | test file(s) could not be found                              |
+| `ERR_MOCHA_UNSUPPORTED`            | requested behavior, option, or parameter is unsupported      |
```
Comment (geshii) [inline docs/index.md:2264]:
```suggestion
| Code                               | Description                                                  |
| ---------------------------------- | ------------------------------------------------------------ |
| `ERR_MOCHA_INVALID_ARG_TYPE`       | Wrong type was passed for a given argument                   |
| `ERR_MOCHA_INVALID_ARG_VALUE`      | Invalid or unsupported value was passed for a given argument |
| `ERR_MOCHA_INVALID_EXCEPTION`      | A falsy or otherwise underspecified exception was thrown     |
| `ERR_MOCHA_INVALID_INTERFACE`      | Interface specified in options not found                     |
| `ERR_MOCHA_INVALID_REPORTER`       | Reporter specified in options not found                      |
| `ERR_MOCHA_NO_FILES_MATCH_PATTERN` | Test file(s) could not be found                              |
| `ERR_MOCHA_UNSUPPORTED`            | Requested behavior, option, or parameter is unsupported      |
```

**Review by geshii — CHANGES_REQUESTED** [review]
minor changes
Sorry for bumping

**Comment by JoshuaKGoldberg** [comment]
This has been sitting with requested changes for a few years now. And it's mostly small cosmetic things that just tinker with the docs, rather than significantly improving them. Closing out as an old PR. 

If someone really wants these changes in, please yell at me and say why. 🙂
